internal/file-detector/detector.go

```diff
@@ -75,7 +75,7 @@ type fsNotify struct {
 	watcher    *fsnotify.Watcher
 }
 
-type fsNotifyOptions struct {
+type FsNotifyOptions struct {
```

Comment:
nit: it's conventional in Go for acronyms to be all uppercase; this should be called `FSNotifyOptions`